File: flux-basic-java/src/main/java/com/example/app/authentication/Sender.java
package com.example.app.authentication;

import com.example.app.user.api.UserId;
import com.fasterxml.jackson.annotation.JsonIgnore;
import io.fluxzero.sdk.tracking.handling.authentication.User;
import lombok.Builder;
import lombok.NonNull;

@Builder(toBuilder = true)
public record Sender(@NonNull UserId userId, Role userRole) implements User {

    public static final Sender system = builder()
            .userId(new UserId("system")).userRole(Role.ADMIN).build();

    public static Sender getCurrent() {
        return User.getCurrent();
    }

    @Override
    @JsonIgnore
    public String getName() {
        return userId.getFunctionalId();
    }

    public boolean hasRole(Role role) {
        return role.matches(userRole);
    }

    @Override
    public boolean hasRole(String role) {
        return hasRole(Role.valueOf(role));
    }

    public boolean isAdmin() {
        return Role.ADMIN.matches(userRole);
    }
}

File: flux-basic-kotlin/src/main/kotlin/com/example/app/authentication/Sender.kt
package com.example.app.authentication

import com.example.app.user.api.UserId
import com.fasterxml.jackson.annotation.JsonIgnore
import io.fluxzero.sdk.tracking.handling.authentication.User

data class Sender(
    val userId: UserId,
    val userRole: Role?
) : User {

    companion object {
        @JvmStatic
        val system = Sender(
            userId = UserId("system"),
            userRole = Role.admin
        )

        @JvmStatic
        fun getCurrent(): Sender {
            return User.getCurrent()
        }
    }

    @JsonIgnore
    override fun getName(): String {
        return userId.functionalId
    }

    fun hasRole(role: Role): Boolean {
        return role.matches(userRole)
    }

    override fun hasRole(role: String): Boolean {
        return hasRole(Role.valueOf(role))
    }

    fun isAdmin(): Boolean {
        return Role.admin.matches(userRole)
    }
}